lib/srv/alpnproxy/proxy.go

```diff
@@ -379,14 +394,54 @@ func (p *Proxy) handleConn(ctx context.Context, clientConn net.Conn) error {
 		return trace.Wrap(err)
 	}
 
+	var handlerConn net.Conn = tlsConn
+	// Check if Multiplex is supported/required by the client.
+	if len(hello.SupportedProtos) > 0 && common.IsPingProtocol(common.Protocol(hello.SupportedProtos[0])) {
```

Comment:
`getHandleDescBasedOnALPNVal` goes through the whole list so `hello.SupportedProtos[0]` here may not match the selected one

Reply:
I've updated this to use the negotiated protocol instead of the protocol list the client provided. Using it guarantees that both sides of the connection are using ping connections.